### ISSUE TITLE
scroll ids getting exhausted quickly in 7.5.2 compared to 7.3.2 with the same scroll context settings

### ISSUE THREAD

**Issue by pkartikk** [elastic/elasticsearch#51865]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.5.2

**Plugins installed**: Default installation

**JVM version** (`java -version`): openjdk version "13.0.1" 2019-10-15

**OS version** (`uname -a` if on a Unix-like system): Linux 4.4.0-87-generic #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**: When we are querying documents from Elasticsearch 7.5.2, the ScrollIds are getting exhausted quickly. The default limit is 500. This limit gets exhausted very quickly with very less load (less than 1000 documents). We haven't observed the same issue with the same API in version 7.3.2 with millions of documents.

```
org.elasticsearch.ElasticsearchStatusException: Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]
        at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:177)
        at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:1727)
        at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:1704)
        at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1467)
        at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1424)
        at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1394)
        at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:930)
        at com.bmc.truesight.saas.ims.es.dao.CIDocRepositoryImpl.getCIDocByTenant(CIDocRepositoryImpl.java:603)
        ... 31 common frames omitted
        Suppressed: org.elasticsearch.client.ResponseException: method [POST], host [machine-name:9200], URI [/ci_index_ims/_search?typed_keys=true&ignore_unavailable=false&expand_wildcards=open&allow_no_indices=true&ignore_throttled=true&scroll=60000ms&search_type=dfs_query_then_fetch&batched_reduce_size=512&ccs_minimize_roundtrips=true], status line [HTTP/1.1 500 Internal Server Error]
{"error":{"root_cause":[{"type":"exception","reason":"Trying to create too many scroll contexts. Must be less than or equal to: [500]. This limit can be set by changing the [search.max_open_scroll_context] setting."}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"dfs","grouped":true,"failed_shards":[{"shard":0,"index":"ci_index_ims","node":"fU1CWJ5NSX6whkIYHQz-sQ","reason":{"type":"exception","reason":"Trying to create too many scroll contexts. Must be less than or equal to: [500]. This limit can be set by changing the [search.max_open_scroll_context] setting."}}]},"status":500}
                at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:253)
                at org.elasticsearch.client.RestClient.performRequest(RestClient.java:231)
                at org.elasticsearch.client.RestClient.performRequest(RestClient.java:205)
                at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1454)
                ... 35 common frames omitted
```

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Settings:

{
   "analysis": {
      "analyzer": {
         "keyword_analyzer": {
            "filter": "lowercase",
            "tokenizer": "keyword"
         }
      }
   }
}

 2. Mapping:

{
              "properties": {
                             "ack": {
                                           "type": "object",
                                           "properties": {
                                                          "slot1": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot2": {
                                                                        "type": "text",
                                                                        "fielddata": true,
                                                                        "fields": {
                                                                                      "keyword": {
                                                                                                     "type": "keyword",
                                                                                                     "ignore_above": 256
                                                                                      }
                                                                        }
                                                          },
                                                          "slot3": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot4": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot5": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot6": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          }
                                           }
                             },
                             "slot7": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot9": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot10": {
                                           "type": "object",
                                           "properties": {
                                                          "slot1": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot2": {
                                                                        "type": "text",
                                                                        "fielddata": true,
                                                                        "fields": {
                                                                                      "keyword": {
                                                                                                     "type": "keyword",
                                                                                                     "ignore_above": 256
                                                                                      }
                                                                        }
                                                          },
                                                          "slot3": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot4": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot5": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot6": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          }
                                           }
                             },
                             "slot12": {
                                           "type": "object",
                                           "properties": {
                                                          "slot1": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot2": {
                                                                        "type": "text",
                                                                        "fielddata": true,
                                                                        "fields": {
                                                                                      "keyword": {
                                                                                                     "type": "keyword",
                                                                                                     "ignore_above": 256
                                                                                      }
                                                                        }
                                                          },
                                                          "slot3": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot4": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot5": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot6": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          }
                                           }
                             },
                             "child_doc_type": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot13": {
                                           "type": "long"
                             },
                             "slot14": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot_join": {
                                           "type": "join",
                                           "eager_global_ordinals": true,
                                           "relations": {
                                                          "slot_parent": [
                                                                        "slot_child_1",
                                                                        "slot_child_2"
                                                          ]
                                           }
                             },
                             "slot15": {
                                           "type": "keyword",
                                           "fields": {
                                                          "ignore_case": {
                                                                        "type": "text",
                                                                        "analyzer": "keyword_analyzer"
                                                          }
                                           }
                             },
                             "slot16": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot17": {
                                           "type": "text",
                                           "fielddata": true
                             },
                             "slot18": {
                                           "type": "text",
                                           "fielddata": true
                             },
                             "slot19": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot20": {
                                           "type": "object",
                                           "properties": {
                                                          "slot1": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot2": {
                                                                        "type": "text",
                                                                        "fielddata": true,
                                                                        "fields": {
                                                                                      "keyword": {
                                                                                                     "type": "keyword",
                                                                                                     "ignore_above": 256
                                                                                      }
                                                                        }
                                                          },
                                                          "slot3": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot4": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot5": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot6": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          }
                                           }
                             },
                             "slot21": {
                                           "type": "long"
                             },
                             "slot22": {
                                           "type": "keyword",
                                           "fields": {
                                                          "ignore_case": {
                                                                        "type": "text",
                                                                        "analyzer": "keyword_analyzer"
                                                          }
                                           }
                             },
                             "slot23": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot24": {
                                           "type": "keyword",
                                           "fields": {
                                                          "ignore_case": {
                                                                        "type": "text",
                                                                        "analyzer": "keyword_analyzer"
                                                          }
                                           }
                             },
                             "slot25": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot26": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot27": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot28": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot29": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot30": {
                                           "type": "text",
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "is_parent": {
                                           "type": "text",
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot48": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot31": {
                                           "type": "long"
                             },
                             "slot32": {
                                           "type": "text",
                                           "fielddata": true
                             },
                             "slot33": {
                                           "type": "object",
                                           "properties": {
                                                          "slot1": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot2": {
                                                                        "type": "text",
                                                                        "fielddata": true,
                                                                        "fields": {
                                                                                      "keyword": {
                                                                                                     "type": "keyword",
                                                                                                     "ignore_above": 256
                                                                                      }
                                                                        }
                                                          },
                                                          "slot3": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot4": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot5": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          },
                                                          "slot6": {
                                                                        "type": "object",
                                                                        "properties": {
                                                                                      "count": {
                                                                                                     "type": "long"
                                                                                      },
                                                                                      "slot_level": {
                                                                                                     "type": "text",
                                                                                                     "fielddata": true,
                                                                                                     "fields": {
                                                                                                                   "keyword": {
                                                                                                                                 "type": "keyword",
                                                                                                                                 "ignore_above": 256
                                                                                                                   }
                                                                                                     }
                                                                                      }
                                                                        }
                                                          }
                                           }
                             },
                             "parent_id": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot11": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot34": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot35": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot36": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot37": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                         }
                                           }
                             },
                             "slot38": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot39": {
                                           "type": "keyword",
                                           "fields": {
                                                          "ignore_case": {
                                                                        "type": "text",
                                                                        "analyzer": "keyword_analyzer"
                                                          }
                                           }
                             },
                             "slot40": {
                                           "type": "keyword",
                                           "fields": {
                                                          "ignore_case": {
                                                                        "type": "text",
                                                                        "analyzer": "keyword_analyzer"
                                                          }
                                           }
                             },
                             "slot41": {
                                           "type": "keyword",
                                           "fields": {
                                                          "ignore_case": {
                                                                        "type": "text",
                                                                        "analyzer": "keyword_analyzer"
                                                          }
                                           }
                             },
                             "slot42": {
                                           "type": "long"
                             },
                             "slot43": {
                                           "type": "long"
                             },
                             "slot44": {
                                           "type": "long"
                             },
                             "slot1": {
                                           "type": "long"
                             },
                             "slot4": {
                                           "type": "long"
                             },
                             "slot5": {
                                           "type": "long"
                             },
                             "slot6": {
                                           "type": "long"
                             },
                             "slot3": {
                                           "type": "long"
                             },
                             "slot45": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             },
                             "slot46": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                         }
                                           }
                             },
                             "slot47": {
                                           "type": "text",
                                           "fielddata": true,
                                           "fields": {
                                                          "keyword": {
                                                                        "type": "keyword",
                                                                        "ignore_above": 256
                                                          }
                                           }
                             }
              }
}

 3. Search:

{
  "from": 0,
  "size": 200,
  "query": {
    "bool": {
      "must": [
        {
          "has_parent": {
            "query": {
              "bool": {
                "must": [
                  {
                    "term": {
                      "slot35.keyword": {
                        "value": "value_0",
                        "boost": 1
                      }
                    }
                  },
                  {
                    "term": {
                      "slot38": {
                        "value": "value_1",
                        "boost": 1
                      }
                    }
                  },
                  {
                    "bool": {
                      "adjust_pure_negative": true,
                      "boost": 1
                    }
                  }
                ],
                "adjust_pure_negative": true,
                "boost": 1
              }
            },
            "parent_type": "slot_parent",
            "score": true,
            "ignore_unmapped": false,
            "boost": 1,
            "inner_hits": {
              "ignore_unmapped": false,
              "from": 0,
              "size": 3,
              "version": false,
              "seq_no_primary_term": false,
              "explain": false,
              "track_scores": false
            }
          }
        },
        {
          "term": {
            "child_doc_type": {
              "value": "value_2",
              "boost": 1
            }
          }
        },
        {
          "bool": {
            "should": [
              {
                "range": {
                  "slot33.slot5.count": {
                    "from": 0,
                    "to": null,
                    "include_lower": false,
                    "include_upper": true,
                    "boost": 1
                  }
                }
              },
              {
                "range": {
                  "slot33.slot4.count": {
                    "from": 0,
                    "to": null,
                    "include_lower": false,
                    "include_upper": true,
                    "boost": 1
                  }
                }
              },
              {
                "range": {
                  "slot33.slot6.count": {
                    "from": 0,
                    "to": null,
                    "include_lower": false,
                    "include_upper": true,
                    "boost": 1
                  }
                }
              },
              {
                "range": {
                  "slot33.slot1.count": {
                    "from": 0,
                    "to": null,
                    "include_lower": false,
                    "include_upper": true,
                    "boost": 1
                  }
                }
              }
            ],
            "adjust_pure_negative": true,
            "boost": 1
          }
        }
      ],
      "adjust_pure_negative": true,
      "boost": 1
    }
  },
  "sort": [
    {
      "slot1": {
        "order": "desc"
      }
    },
    {
      "slot4": {
        "order": "desc"
      }
    },
    {
      "slot5": {
        "order": "desc"
      }
    },
    {
      "slot6": {
        "order": "desc"
      }
    },
    {
      "slot3": {
        "order": "desc"
      }
    },
    {
      "slot33.slot1.count": {
        "order": "desc"
      }
    },
    {
      "slot33.slot4.count": {
        "order": "desc"
      }
    },
    {
      "slot33.slot5.count": {
        "order": "desc"
      }
    },
    {
      "slot33.slot6.count": {
        "order": "desc"
      }
    },
    {
      "slot33.slot3.count": {
        "order": "desc"
      }
    },
    {
      "slot24": {
        "order": "asc"
      }
    }
  ]
}

**Provide logs (if relevant)**:
 We see the following exception:

```
org.elasticsearch.transport.RemoteTransportException: [machine_name][machine IP:9300][indices:data/read/search[phase/query]]
Caused by: org.elasticsearch.ElasticsearchException: Trying to create too many scroll contexts. Must be less than or equal to: [500]. This limit can be set by changing the [search.max_open_scroll_context] setting.
             at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:542) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:348) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:340) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:773) ~[elasticsearch-7.5.2.jar:7.5.2]
             at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.5.2.jar:7.5.2]
             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
             at java.lang.Thread.run(Thread.java:830) [?:?]
```

**Comment by jimczi** [comment]
Sorry for the late reply. Nothing has changed regarding the handling of scroll context between 7.3 and 7.5. We fixed a [race condition](https://github.com/elastic/elasticsearch/pull/53449) with the counting of scroll contexts but that was released in `7.6.2`. Note that the number of documents in the index doesn't make a difference here, we always count each shard in the scroll context as an increment of 1. 
I advise you to look at the number of open contexts per node and to check that you delete the scroll when it is exhausted. 
I am closing this issue because we reserve github for verified bugs and feature requests. If you still see the problem please open a discussion in the [forum](https://discuss.elastic.co/c/elastic-stack/elasticsearch/6) where it will be easier to diagnose.

**Comment by jamshid** [comment]
I know this is old and 7.5 is EOL but to be clear, Elasticsearch 7.5 and earlier have a race condition bug around scrolls? 
An attempt was made to fix the race condition in 7.6 but that fix introduced other problems, all fixed in 7.7.
